private/pkg/protoencoding/strip_legacy_options.go

```diff
@@ -0,0 +1,298 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protoencoding
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+// stripLegacyOptions modifies the given slice so that it does not contain legacy
+// elements that the Go protobuf runtime does not support -- namely messages that use
```

Comment:
nit: `%s/modifies the given slice so that it does not contain.../modifies the given slice of file descriptors so that the slice does not contain...`

Just for clarity that `FileDescriptorProto`s is the structure being modified (similar to how subsequent functions make it clear we are modifying file/message/field descriptors).

Reply:
done